jbake-core/src/main/java/org/jbake/app/Renderer.java

```diff
@@ -344,7 +293,40 @@ private DocumentModel buildSimpleModel(String type) {
         return content;
     }
 
-    private interface RenderingConfig {
+    public void addError(Exception e) {
+        errors.add(e);
+    }
+
+    public List<Throwable> getErrors() {
+        return errors;
+    }
+
+    public void shutdown() throws InterruptedException {
+        this.excutor.shutdown();
+        this.excutor.awaitTermination(10, TimeUnit.MINUTES);
+    }
```

Comment:
Ah, you made it synchronous. Why wait here? Better name it "shutdownAndWait" then, because usually the shutdown method does not wait.